sdk/core/Azure.Core/src/LongRunningOperation.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure
+{
+    public abstract class LongRunningOperation<T>: IDisposable
+    {
+        public abstract Response Raw { get; }
+        public abstract ValueTask<Response<T>> GetValueAsync(CancellationToken cancellationToken = default);
+        public abstract Response<T> GetValue(CancellationToken cancellationToken = default);
+
+        public abstract ValueTask<Response<LongRunningOperationStatus>> GetStatusAsync(CancellationToken cancellationToken = default);
+        public abstract Response<LongRunningOperationStatus> GetStatus(CancellationToken cancellationToken = default);
+
+        public abstract ValueTask<Response> CancelAsync(CancellationToken cancellationToken = default);
+        public abstract Response Cancel(CancellationToken cancellationToken = default);
+        public abstract void Dispose();
+    }
+}
```

Comment:
In addition, I have an extension class with methods that allow:
```c#
string value = await client.StartOperationAsync().WaitCompletionAsync();
```

In other words, we agreed at the arch meeting, we will support automatic polling. The polling interval will be by based on some headers (supposedly services send such header), or based on service documentation, or user specified.

Reply:
This is how GetValue behaves in my prototype, it pools until the value is available.